ui/core/src/utils/fetch.ts

```diff
@@ -42,3 +49,13 @@ export class FetchError extends Error {
     Object.setPrototypeOf(this, FetchError.prototype);
   }
 }
+
+/**
+ * General error type for an error that has a message that is OK to show to the end user.
+ */
+export class UserFriendlyError extends Error {
```

Comment:
The error messages from the backend often aren't particularly user friendly right now. Is this aspirational assuming that the backend will be updated eventually to make them more friendly? I'm cool with the change either way to surface errors more helpfully, just curious about the thoughts about the UX going forward.